### PR TITLE
API can add new articles to database

### PR DESCRIPTION
**Pull request by LucasKnudsen**
[**PT-Story:**  ](https://www.pivotaltracker.com/story/show/177157820)
### User Story 
``` 
As an API
In order to be able to create new articles
I need functionality to add new articles to my database
``` 
## Changes proposed in this pull request: 
* Adds an Admin model through devise token auth
* Configures create action for articles
## What I have learned working on this feature: 
* Deeper customization with Rails
* Working with an admin model
## Packages/Gems added 
* devise-token-auth

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Article < ApplicationRecord
   enum article_type: [:experience, :story]
-
   validates_presence_of :article_type, :title, :body, :teaser, :location, :category
+  belongs_to :admin, optional: true
 end
```
Comment (LucasKnudsen) [inline app/models/article.rb:5]: Validate datatype Array for body attribute
Need help with this

File: app/models/article.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Article < ApplicationRecord
   enum article_type: [:experience, :story]
-
   validates_presence_of :article_type, :title, :body, :teaser, :location, :category
+  belongs_to :admin, optional: true
```
Comment (LucasKnudsen) [inline app/models/article.rb:4]: Optional true? 
Can this be fixed without having to revise every test?

File: app/controllers/api/articles_controller.rb [hunk 3]
```diff
@@ -49,4 +49,20 @@ def show
       message: 'Article not found.'
     }, status: 404
   end
+
+  def create
+    current_admin.articles.create(
+      title: params[:title], teaser: params[:teaser], body: params[:body],
+      article_type: params[:article_type], location: params[:location], category: params[:category]
+    )
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb:57]: This could've been done much easier by permitting the params that an article should have.

File: app/controllers/api/articles_controller.rb [hunk 4]
```diff
@@ -49,4 +49,20 @@ def show
...
+    if current_admin.articles.last.persisted?
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb:58]: I would have stored the article you create on line 54 in a variable and then checked if that specific variable persisted

File: app/controllers/api/articles_controller.rb [hunk 5]
```diff
@@ -49,4 +49,20 @@ def show
...
+      render json: {
+        message: 'The article was successfully created.'
+      }, status: 201
+    else
+      render json: {
+        message: 'Please fill out all fields.'
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb:64]: I would've used the validations and rendered more dynamic error message depending on what is missing to save the article

File: app/controllers/api/articles_controller.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 class Api::ArticlesController < ApplicationController
+  before_action :authenticate_admin!, only: [:create]
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb:2]: for me this is not a enough, an editor should not be able to create an article, that is for the journalist. So just setting an admin model is not enough authorization in my opinion

File: db/migrate/20210307093403_change_articles_body_type_to_array.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class ChangeArticlesBodyTypeToArray < ActiveRecord::Migration[6.1]
+  def change
+    change_column :articles, :body, :text, array: true, using: 'body::text[]'
```
Comment (emtalen) [inline db/migrate/20210307093403_change_articles_body_type_to_array.rb:3]: this is out of scope for this user story

File: spec/requests/admin/create_article_spec.rb [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    end
+
+    it 'responds with a 401' do
+      expect(response).to have_http_status 401
+    end
+
+    it 'is expected to not create an article' do
+      articles = Article.all
+      expect(articles.count).to eq 0
+    end
```
Comment (emtalen) [inline spec/requests/admin/create_article_spec.rb:87]: again, because I have commented that before, test for the message in the response
